driver-core/src/test/functional/com/mongodb/internal/operation/AsyncQueryBatchCursorFunctionalSpecification.groovy

```diff
@@ -357,7 +357,7 @@ class AsyncQueryBatchCursorFunctionalSpecification extends OperationFunctionalSp
         while (connection.getCount() > 1) {
             Thread.sleep(5)
         }
-        makeAdditionalGetMoreCall(getNamespace(), firstBatch.cursor, connection as Connection)
+        makeAdditionalGetMoreCall(getNamespace(), firstBatch.cursor, new SyncConnection(connection))
```

Comment:
It's because the code was not correct: `connection` is of the type `AsyncConnection`, but it was casted to `Connection`. Groovy's lack of static typing allowed this code to be compiled (a language with static typing would have reported an error at compile time), and since the run-time type used to be `TestConnection`, the code was working. This PR changed the run-time type of `connection`, while preserving its compile-time type, which resulted in a run-time exception.